internal/controller/replication.storage/volumereplication_controller.go

```diff
@@ -371,7 +371,7 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	default:
 		replicationErr = fmt.Errorf("unsupported volume state")
 		logger.Error(replicationErr, "given volume state is not supported", "ReplicationState", instance.Spec.ReplicationState)
-		setFailureCondition(instance)
+		setFailureCondition(instance, "unsupported volume state", replicationErr.Error(), instance.Spec.DataSource.Kind)
```

Comment:
use `replicationErr.Error() here?

Reply:
yes, we dont have any other error here to be passed.